src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -868,3 +868,248 @@ def describe_non_ready_agent_log(filepath_with_timestamp, corev1_api_instance, a
+
+    try:
+        if storage_space_available:
+            command = [kubectl_client_location, "get", "secrets", "-n", "azure-arc"]
+            if kube_config:
+                command.extend(["--kubeconfig", kube_config])
+            if kube_context:
+                command.extend(["--context", kube_context])
+            # Using Popen to execute the command and fetching the output
+            response_kubectl_get_events = Popen(command, stdout=PIPE, stderr=PIPE)
+            output_kubectl_get_events, error_kubectl_get_events = response_kubectl_get_events.communicate()
+            if response_kubectl_get_events.returncode != 0:
+                telemetry.set_exception(exception=error_kubectl_get_events.decode("ascii"), fault_type=consts.Kubectl_Get_Secrets_Failed_Fault_Type, summary='Error while doing kubectl get secrets')
+                logger.warning("Error while doing kubectl get secrets for azure-arc namespace. We were not able to capture this log in arc_diganostic_logs folder. Exception: ", error_kubectl_get_events.decode("ascii"))
+                diagnoser_output.append("Error while doing kubectl get secrets in azure-arc namespace. We were not able to capture this log in arc_diganostic_logs folder. Exception: " + error_kubectl_get_events.decode("ascii"))
+                return storage_space_available
+
+            # Converting output obtained in json format
+            events_json = output_kubectl_get_events.decode()
+            # Path to add the azure-arc secrets
+            event_logs_path = os.path.join(filepath_with_timestamp, "azure-arc-secrets.txt")
+
+            with open(event_logs_path, 'w+') as event_log:
+                    event_log.write(events_json)
+
+            return storage_space_available
+
+    # For handling storage or OS exception that may occur during the execution
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            storage_space_available = False
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+        else:
+            logger.warning("An exception has occured while storing azure arc secrets in the user local machine. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Kubectl_Get_Secrets_Failed_Fault_Type_Exception, summary="Exception occured while storing azure arc secrets")
+            diagnoser_output.append("An exception has occured while storing azure arc secrets in the user local machine. Exception: {}".format(str(e)) + "\n")
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+            logger.warning("An exception has occured while storing azure arc secrets in the user local machine. Exception: {}".format(str(e)) + "\n")
```

Comment:
let's say "storing list of secret names" instead of secrets. User might think we're actually storing secrets